ntripclient.c

```diff
@@ -1636,24 +1696,60 @@ int main(int argc, char **argv)
                   starttime = time(0);
                   lastout = starttime;
                 }
-                if(args.serdevice && !stop)
+
+                // get subsequent NMEA GGA sentence from GGA source and
+                // send it to NTRIP caster.  Source can be either serial device
+                // or IPC pipe
+                if(args.nmea && !stop)
                 {
                   int doloop = 1;
                   while(doloop && !stop)
                   {
-                    int i = SerialRead(&sx, buf, 200);
+                    int i = 0;
+                    // get NMEA GGA sentence from serial device or IPC pipe
+                    if (args.serdevice) {
+                      i = SerialRead(&sx, buf, 200);
+                    } else {
+
+#ifdef NTRIPCLIENT_DEBUG_LOG
+                      fprintf(flogfile, "requesting IPC receive\n");
+#endif // NTRIPCLIENT_DEBUG_LOG
+
+                      i = fread(buf, 1, 200, stdin);
+
+#ifdef NTRIPCLIENT_DEBUG_LOG
+                      fprintf(flogfile, "IPC receive %d bytes\n", i);
+                      int k = 0;
+                      for (k = 0;  k < i; k++) {
+                        fprintf(flogfile, "%c", buf[k]);
+                      }
+                      fprintf(flogfile, "\n");
+#endif // NTRIPCLIENT_DEBUG_LOG
+
+                    }
                     if(i < 0)
                     {
-                      fprintf(stderr, "Could not access serial device\n");
+#ifdef NTRIPCLIENT_DEBUG_LOG
+                      fprintf(flogfile, "Could not get NMEA GGA data\n");
+#endif // NTRIPCLIENT_DEBUG_LOG
+
+                      fprintf(stderr, "Could not get NMEA GGA data\n");
                       stop = 1;
                     }
                     else
                     {
                       int j = 0;
                       if(i < 200) doloop = 0;
+
+                      // TODO: is this needed??? (write to IPC pipe back to MATE)
                       fwrite(buf, i, 1, stdout);
```

Comment:
@zalice @adriennet823 @AkshayMatternet @bhiln @AydenChubbic 
This line was in the original open source version. This will send the GGA sentence (that we just got from MATE over the pipe) back to MATE (via the other pipe).  But is this necessary?  Does MATE need the GGA sent to it? I don't think it does, especially since it just sent us this GGA sentence.

Reply:
MATE does not need the GGA sent back to it. I'm curious what the use-case of this was in the original implementation🤔

Reply:
My best guess is that the original use case was that this was actually sent to stdout as in the screen for someone to monitor.

Reply:
I have removed this line.